### PR TITLE
Optimize NSString.lineAndCharacterForByteOffset()

### PR DESCRIPTION
**Pull request by norio-nomura**
Add `NSString.CacheContainer.lineAndCharacterForByteOffset(_:)`

This commit reduces the duration of linting KeychainAccess by SwiftLint (8d529f1) from:
```
swiftlint lint  27.27s user 0.25s system 96% cpu 28.424 total
```
to:
```
swiftlint lint  3.76s user 0.10s system 89% cpu 4.291 total
```

This is that I mentioned in #127

### CODE REVIEW

**Comment by jpsim** [comment]
Ouch, this was a heavy one. I'd like to integrate some really rough estimate performance tests, but with tests running on a wide variety of systems, I'm not sure it's possible to do reliably. Do you have any ideas?

File: Source/SourceKittenFramework/String+SourceKitten.swift [hunk 1]
```diff
@@ -101,14 +120,22 @@ extension NSString {
...
+                let element = rangesAndLine[$0]
+                return (line: element.line.index, character: offset - element.range.location + 1)
+            }
+        }
+        
+        func lineAndCharacterForByteOffset(offset: Int) -> (line: Int, character: Int)? {
+            let index = rangesAndLine.indexOf { NSLocationInRange(offset, $0.byteRange) }
+            return index.map {
+                let element = rangesAndLine[$0]
+                return (line: element.line.index, character: offset - element.range.location + 1)
```
Comment (norio-nomura) [inline Source/SourceKittenFramework/String+SourceKitten.swift]: @jpsim: Sorry, I have a bug here. The `character` is not correct. I'll fix that.

Comment (norio-nomura) [inline Source/SourceKittenFramework/String+SourceKitten.swift]: pushed: https://github.com/jpsim/SourceKitten/pull/129

**Comment by norio-nomura** [comment]
Testing on CI? I don't have enough knowledge about that.
I know that Xcode has performance test feature triggering by performance regressions. But that needs setting base performance thresholds on real machine and may not fit CI services.

**Pull request merged**